lms/validation/_bearer_token.py

```diff
@@ -0,0 +1,219 @@
+"""Schema for our bearer token-based LTI authentication."""
+import datetime
+
+import jwt
+from jwt.exceptions import ExpiredSignatureError, InvalidTokenError
+from webargs.pyramidparser import parser
+import marshmallow
+from pyramid.httpexceptions import HTTPUnprocessableEntity
+
+from lms.validation._exceptions import (
+    ValidationError,
+    ExpiredSessionTokenError,
+    MissingSessionTokenError,
+    InvalidSessionTokenError,
+)
+from lms.values import LTIUser
+
+
+__all__ = ("BearerTokenSchema",)
+
+
+class BearerTokenSchema(marshmallow.Schema):
+    """
+    Schema for our bearer token-based LTI authentication.
+
+    Serializes :class:`~lms.values.LTIUser` objects into signed and
+    timestamped ``"Bearer <ENCODED_JWT>"`` strings, and deserializes these
+    bearer strings back into :class:`~lms.values.LTIUser` objects. The
+    JWT signature and timestamp are verified during deserialization.
+
+    Usage:
+
+        >>> schema = BearerTokenSchema(request)
+
+        >>> # Serialize an LTIUser (``request.lti_user`` in this example)
+        >>> # into an authorization param value:
+        >>> schema.authorization_param(request.lti_user)
+        'Bearer eyJ...YoM'
+
+        >>> # Deserialize the request's authorization param into an LTI user.
+        >>> schema.lti_user()
+        LTIUser(user_id='...', oauth_consumer_key='...', ...)
+
+    The above are convenience methods that wrap webargs and marshmallow. But
+    this class is also a marshmallow schema and can be used via the usual
+    webargs or marshmallow APIs. For example to serialize a dict and then
+    deserialize the same dict back from its serialized form using marshmallow::
+
+        >>> schema.dump(request.lti_user).data
+        {'authorization': 'Bearer eyJ...YoM'}
+
+        >>> schema.load({'authorization': 'Bearer eyJ...YoM'}).data
+        LTIUser(user_id='...', oauth_consumer_key='...', ...)
+
+    Or to parse an :class:`~lms.values.LTIUser` out of a Pyramid
+    request's authorization param using webargs::
+
+        >>> from webargs.pyramidparser import parser
+        >>> parser.parse(s, request)
+        LTIUser(user_id='...', oauth_consumer_key='...', ...)
+    """
+
+    user_id = marshmallow.fields.Str(required=True)
+    oauth_consumer_key = marshmallow.fields.Str(required=True)
+
+    class Meta:
+        """Marshmallow options for this schema."""
+
+        # Silence a strict=False deprecation warning from marshmallow.
+        # TODO: Remove this once we've upgraded to marshmallow 3.
+        strict = True
```

Comment:
I guess this is about https://github.com/marshmallow-code/marshmallow/issues/377?

Reply:
Yes. `strict=False` is the default in marshmallow 2. In the upcoming v3 strict is always on and the strict option has been removed. Recent 2.x versions emit a warning if you _don't_ explicitly set `strict=True`